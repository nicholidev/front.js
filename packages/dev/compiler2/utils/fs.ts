import { promises as fsp } from "fs";
import * as path from "path";

export async function writeFileSafe(
  file: string,
  contents: string
): Promise<string> {
  await fsp.mkdir(path.dirname(file), { recursive: true });
  await fsp.writeFile(file, contents);
  return file;
}

export async function writeFilesSafe(
  files: { file: string; contents: string }[]
): Promise<string[]> {
  return Promise.all(
    files.map(({ file, contents }) => writeFileSafe(file, contents))
  );
}

export async function createTemporaryDirectory(
  baseDir: string
): Promise<string> {
  return fsp.mkdtemp(path.join(baseDir, "remix-"));
}
