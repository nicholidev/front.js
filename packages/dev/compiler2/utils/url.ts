import * as path from "path";

export function createUrl(publicPath: string, file: string): string {
  return publicPath + file.split(path.win32.sep).join("/");
}
