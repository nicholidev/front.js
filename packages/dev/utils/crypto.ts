import type { BinaryLike } from "crypto";
import { createHash } from "crypto";

export function getHash(source: BinaryLike): string {
  return createHash("sha1").update(source).digest("hex");
}
