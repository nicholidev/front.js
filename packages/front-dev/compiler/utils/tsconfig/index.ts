import tsConfigPaths from "tsconfig-paths";

export function createMatchPath(tsconfigPath: string | undefined) {
  // There is no tsconfig to match paths against.
  if (!tsconfigPath) {
    return undefined;
  }

  // When passing a absolute path, loadConfig assumes that the path contains
  // a tsconfig file.
  // Ref.: https://github.com/dividab/tsconfig-paths/blob/v4.0.0/src/__tests__/config-loader.test.ts#L74
  let configLoaderResult = tsConfigPaths.loadConfig(tsconfigPath);

  if (configLoaderResult.resultType === "failed") {
    if (configLoaderResult.message === "Missing baseUrl in compilerOptions") {
      throw new Error(
        `ðŸš¨ Oops! No baseUrl found, please set compilerOptions.baseUrl in your tsconfig or jsconfig`
      );
    }
    return undefined;
  }

  return tsConfigPaths.createMatchPath(
    configLoaderResult.absoluteBaseUrl,
    configLoaderResult.paths,
    configLoaderResult.mainFields,
    configLoaderResult.addMatchAll
  );
}
