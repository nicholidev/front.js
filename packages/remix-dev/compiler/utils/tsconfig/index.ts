import tsConfigPaths from "tsconfig-paths";

import { loadTsConfig } from "./configLoader";
export { loadTsConfig } from "./configLoader";

export function createMatchPath() {
  let configLoaderResult = loadTsConfig();
  if (configLoaderResult.resultType === "failed") {
    return undefined;
  }

  let matchPath = tsConfigPaths.createMatchPath(
    configLoaderResult.absoluteBaseUrl,
    configLoaderResult.paths,
    configLoaderResult.mainFields,
    configLoaderResult.addMatchAll
  );

  return matchPath;
}
