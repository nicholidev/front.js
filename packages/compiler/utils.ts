import fs from "fs";
import path from "path";

export function readVendorFile(filename: string): string {
  let file = path.resolve(__dirname, "./vendor", filename);
  return fs.readFileSync(file, "utf-8");
}
