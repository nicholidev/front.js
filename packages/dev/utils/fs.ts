import { promises as fsp } from "fs";
import * as path from "path";
// @ts-expect-error
import readPackageJson from "read-package-json-fast";

export async function writeFileSafe(
  file: string,
  contents: string
): Promise<void> {
  await fsp.mkdir(path.dirname(file), { recursive: true });
  await fsp.writeFile(file, contents);
}

export async function createTemporaryDirectory(
  baseDir: string
): Promise<string> {
  return fsp.mkdtemp(path.join(baseDir, "remix-"));
}
