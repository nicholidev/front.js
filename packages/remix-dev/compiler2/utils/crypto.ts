import * as fs from "fs";
import type { BinaryLike } from "crypto";
import { createHash } from "crypto";

export function getHash(source: BinaryLike): string {
  return createHash("sha256").update(source).digest("hex");
}

export async function getFileHash(file: string): Promise<string> {
  return new Promise((accept, reject) => {
    let hash = createHash("sha256");
    fs.createReadStream(file)
      .on("error", error => reject(error))
      .on("data", data => hash.update(data))
      .on("close", () => {
        accept(hash.digest("hex"));
      });
  });
}
