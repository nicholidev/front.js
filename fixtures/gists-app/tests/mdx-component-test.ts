import type { Browser, Page } from "puppeteer";
import puppeteer from "puppeteer";

import { getHtml, reactIsHydrated } from "./utils";

const testPort = 3000;
const testServer = `http://localhost:${testPort}`;

describe("mdx component", () => {
  let browser: Browser;
  let page: Page;
  beforeEach(async () => {
    browser = await puppeteer.launch();
    page = await browser.newPage();
  });

  afterEach(() => browser.close());

  it("works", async () => {
    await page.goto(`${testServer}/page/one`);
    await reactIsHydrated(page);

    expect(
      await getHtml(page, '[data-test-id="counter-button"]')
    ).toMatchSnapshot();

    await page.click('[data-test-id="counter-button"]');

    expect(
      await getHtml(page, '[data-test-id="counter-button"]')
    ).toMatchSnapshot();
  });
});
