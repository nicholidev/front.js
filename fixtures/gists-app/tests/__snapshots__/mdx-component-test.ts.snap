// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mdx component works 1`] = `
"<button data-test-id=\\"counter-button\\">Clicked 0</button>
"
`;

exports[`mdx component works 2`] = `
"<button data-test-id=\\"counter-button\\">Clicked 1</button>
"
`;
